auth.go

```diff
@@ -0,0 +1,151 @@
+package main
+
+import (
+	"bufio"
+	"encoding/json"
+	"fmt"
+	"os"
+	"os/exec"
+	"strings"
+	"time"
+
+	"freechatgpt/internal/tokens"
+
+	"github.com/acheong08/OpenAIAuth/auth"
+)
+
+var accounts []Account
+
+type Account struct {
+	Email    string `json:"username"`
+	Password string `json:"password"`
+}
+
+// Read accounts.txt and create a list of accounts
+func readAccounts() {
+	accounts = []Account{}
+	// Read accounts.txt and create a list of accounts
+	if _, err := os.Stat("accounts.txt"); err == nil {
+		// Each line is a proxy, put in proxies array
+		file, _ := os.Open("accounts.txt")
+		defer file.Close()
+		scanner := bufio.NewScanner(file)
+		for scanner.Scan() {
+			// Split by :
+			line := strings.Split(scanner.Text(), ":")
+			// Create an account
+			account := Account{
+				Email:    line[0],
+				Password: line[1],
+			}
+			// Append to accounts
+			accounts = append(accounts, account)
+		}
+	}
+}
+func scheduleToken() {
```

Comment:
there are a lot of auth rate limits. how about renewing 1 token every 10 minutes and keep that running indefinitely?

Reply:
This renew token func generate a new accesstoken, means a new login. every 10 minutes 1 new login may be too frequently. 
If you mean extend the validity period of the current accesstoken, that need http get `https://chat.openai.com/api/auth/session` with cookies, means the package OpenAIAuth func partSix needs rerun every 10 minutes and need keep session state(include cookies).

Reply:
Hmm. It kinda depends how many accounts you have. For something like 500 accounts it might not be unreasonable. Not sure how to calculate. Delaying everything till there is nearly no time left might not be optimal either. I'll merge first and deal with it later ig